presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -302,11 +313,25 @@ private void invokeNoMoreSplitsIfNecessary()
     private ListenableFuture<?> loadPartition(HivePartitionMetadata partition)
             throws IOException
     {
-        String partitionName = partition.getHivePartition().getPartitionId();
+        HivePartition hivePartition = partition.getHivePartition();
+        String partitionName = hivePartition.getPartitionId();
         Properties schema = getPartitionSchema(table, partition.getPartition());
         List<HivePartitionKey> partitionKeys = getPartitionKeys(table, partition.getPartition());
         TupleDomain<HiveColumnHandle> effectivePredicate = compactEffectivePredicate.transform(HiveColumnHandle.class::cast);
 
+        List<HiveColumnHandle> partitionColumns = getPartitionKeyColumnHandles(table, typeManager);
+        BooleanSupplier partitionMatchSupplier = () -> {
+            TupleDomain<ColumnHandle> dynamicFilter = dynamicFilterSupplier.get();
+            if (dynamicFilter.isNone()) {
```

Comment:
Currently `partitionMatches` doesn't handle filter is none case. I didn't add this to `partitionMatches` function because in the existing use of this function in HivePartitionManager, none handling was not needed because it was handled separately in getPartitions.

Reply:
> Currently partitionMatches doesn't handle filter is none case.

Let's add support for `none` to `partitionMatches` as this would make `partitionMatches` complete

Reply:
Done